### DISCUSSION TITLE
Ambisonic audio files in Spoke?

### DISCUSSION THREAD

**Discussion by gantttt** [mozilla/hubs#3187] [Spoke Questions] [answered]
Hey gang! 

Super curious: Does anyone have any wisdom re: ambisonic audio files in Spoke? (or is this even possible?) Would love to be able to drop an ambiX/b format sound environmoent in as opposed to working with point sources, but no luck so far. Any clever workarounds i wonder? (maybe a 360 video that's 'audio only')?

I've had some luck using a 3rd party ambisonics component in A-frame, but not sure how (or if) this would translate to Hubs/Spoke. 

thanks a bundle!

**Comment by rawnsley** [comment, marked as answer]
If there is an A-frame component (or any web-based component) that supports ambisonics then it is certainly possible, but you would need to add that component to a custom client (see Hubs Cloud for more information). You could petition the developers to add it or submit a Git Pull Request with the proposed changes and they may decide to include it into the main branch of Hubs, in which case it would work for all Hubs rooms.

Alternatively you could split the sound file into individual mono audio files using something like [Spatial Workstation](https://facebook360.fb.com/spatial-workstation/) and add each one individually into a scene in Spoke.

**Reply by gantttt** [reply]
Thanks @rawnsley + beautifully said! 

Agree: was able to get [this ](https://github.com/datavized/aframe-ambisonic-component)custom component working in A-frame, but a little scared of adding it to a Hubs Cloud build (though, fortunately, i already have a Cloud set up, so reckon its worth exploring) 

Great call re: requesting the addition though (dev's if your listening - pretty please, ha!) 

Would be curious to hear yr take on this actually: my preference would be to go the 'individual-mono-stems-as-point-sources' approach mentioned above, but I've been having issues with synchronization -- it seems like with longer files/slower connections, they often start at slightly different times, ruining the piece:

This would require addressing a separate issue, but wondering if anyone has had any luck with that approach/problem?

thanks a bundle!
m

**Reply by sobralfreenat** [reply]
"Alternatively you could split the sound file into individual mono audio files using something like [Spatial Workstation](https://facebook360.fb.com/spatial-workstation/) and add each one individually into a scene in Spoke."

This would be great since syncing between individual mono were feasible. Even though this is resolved it would to have to get trancoded to multichanel format like Quad, 5.0, 5.1, 7.1, etc.

Unless we have audio adressed from video to audio source elements and the video src has  multichannel audio embeded already.
